### PR TITLE
[BEAM-6302] Allow setting Compression Codec in ParquetIO

### PR DESCRIPTION
**Pull request by lgajowy**
It's about time to start implementing upgrades to ParquetIO! This small PR enables changing compression from the default one (UNCOMPRESSED) while writing files.

R: @aromanenko-dev 

Could you take a look? Thanks! 

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by lgajowy** [comment]
Run Java PreCommit

File: sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java [hunk 1]
```diff
@@ -262,19 +264,36 @@ public static Sink sink(Schema schema) {
     @Nullable
     abstract String getJsonSchema();
 
+    @Nullable
```
Comment (iemejia) [inline sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java]: You can remove this `@Nullable` by setting the `CompressionCodecName.UNCOMPRESSED` when you build the object `write, sink`. This way you avoid also the `if == null` comparison .

File: sdks/java/io/parquet/src/test/java/org/apache/beam/sdk/io/parquet/ParquetIOTest.java [hunk 2]
```diff
@@ -89,14 +90,23 @@ public void testWriteAndRead() {
 
   @Test
   public void testWriteAndReadFiles() {
+    writeReadAssert(CompressionCodecName.UNCOMPRESSED);
```
Comment (iemejia) [inline sdks/java/io/parquet/src/test/java/org/apache/beam/sdk/io/parquet/ParquetIOTest.java]: Remove this test, it is useless if you set the default I proposed above.

File: sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java [hunk 3]
```diff
@@ -262,19 +264,36 @@ public static Sink sink(Schema schema) {
...
     @AutoValue.Builder
     abstract static class Builder {
       abstract Builder setJsonSchema(String jsonSchema);
 
+      abstract Builder setCompressionCodec(CompressionCodecName compressionCodec);
+
       abstract Sink build();
     }
 
+    /** Specifies compression codec. By default, CompressionCodecName.UNCOMPRESSED. */
```
Comment (iemejia) [inline sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java]: The default should be probably SNAPPY to be consistent with AvroIO (as discussed in slack).

File: sdks/java/io/parquet/src/test/java/org/apache/beam/sdk/io/parquet/ParquetIOTest.java (hunk 2 above)
Comment (lgajowy) [reply sdks/java/io/parquet/src/test/java/org/apache/beam/sdk/io/parquet/ParquetIOTest.java]: ok

File: sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java (hunk 3 above)
Comment (lgajowy) [reply sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java]: ok

File: sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java (hunk 1 above)
Comment (lgajowy) [reply sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java]: ok

**Comment by lgajowy** [comment]
Done! Thanks @iemejia! Could you make another round?

**Comment by aromanenko-dev** [comment]
@lgajowy Thanks for adding this feature, I believe it should be very useful.
@iemejia Thanks for taking care of this PR.

**Pull request merged**